### DISCUSSION TITLE
Multiple Unique Keys - Composite Keys

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#14158] [Feature Requests]
Consider this: I have a user table and a table for (allergic) markings and a m2m junction table for them.

A user could add a marking twice. Therefore a possibility to check if these two parameters (user_id and marking_id) are together unique would be nice. One Possibility would be composite keys.

https://github.com/directus/directus/issues/455
https://github.com/directus/directus/discussions/12137 
https://github.com/directus/directus/discussions/13673

**Comment by NilsBaumgartner1994** [comment]
Is it this field?
Since the description does not give a good hint.

<img width="979" alt="Bildschirmfoto 2022-06-28 um 17 27 36" src="https://user-images.githubusercontent.com/33295329/176219062-be10e2d5-50f1-4198-a2ca-fc5a463bff43.png">

**Reply by rijkvanzanten** [reply]
Nope, that's to control what fields are copied when a duplicate of an item is made

**Comment by claytongulick** [comment]
I’m scratching my head on this a bit too… I have collection that are related by multiple fields - for example, a task has a client_id and a patient_id.

A Patient also has a client_id.

The client_id for a task can’t be different than the client_id for the patient. It exists on the task though because not all tasks necessarily have a patient assigned.

It’s super nice to be able to bring up a patient record and see all their tasks via O2M, and also super nice to be able to create a task for the patient right there by clicking on “Create”!

The less nice part is that while the patient_id is filled in correctly in that scenario, the client_id isn’t. It feels silly to the user to have to manually select the client_id for a task for a patient that already has a client_id assigned.

I did try using {{patient_id.client_id}} as a dynamic variable in the default value, but didn’t have any luck.

In order to make this work, I think there’d need to be some sort of way to indicate a multi-column relationship when specifying the join.

**Comment by mattarau** [comment]
I'd like to add that this would be a requirement in order to have fields set as unique in translation tables.

An example where you can't have a unique field with translations is professions, where sometimes the original name is kept and other times the languages are so close that the translation is the same:

| English      | Spanish            | French       | German        | Portuguese         |
|--------------|--------------------|--------------|---------------|--------------------|
| Chef     | Chef           | Chef     | `Koch`          | Chef           |
| Designer | `Diseñador(a)`       | Designer | Designer  | Designer       |
| Programmer   | `Programador(a)` | Programmeur  | Programmierer | `Programador(a)` |

In the first row, only German translates.
In the second row, only Spanish translates.
In the third row, Spanish and Portuguese translations are the same

I'm not sure how exactly this could be achieved, but ideally it would be nice to always have the field that's set to be unique in a translation table to be a composite with the language's FK field.

**Comment by w0kyj** [comment]
Duplicate of #13673
